src/datepicker/datepicker.js

```diff
@@ -682,7 +683,28 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
         if (value) {
           scope.$broadcast('datepicker.focus');
           scope.position = appendToBody ? $position.offset(element) : $position.position(element);
-          scope.position.top = scope.position.top + element.prop('offsetHeight');
+
+          var clearPosition = function() {
+            scope.position.top = scope.position.bottom = scope.position.left = scope.position.right = 'initial';
+          };
+
+          if(!scope.forcePosition || scope.forcePosition === 'left-bottom') {
```

Comment:
Also this is more semantics, but `bottom-left`, `bottom-right`, `top-right`, and `top-left` are better strings to use.